### DISCUSSION TITLE
All search to work across Element Types.  Particularly Entries & Products

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8128] [Ideas]
> Created by: Jeremy Daalder (jd@imagescience.com.au) on 2015/08/17 22:59:59 +0000
> Votes at time of UserVoice import: 19

-----------------------------------

Currently it's not possible to search everything (i.e. different Element Types) all at once without running multiple search queries.  Also, without access to the score value, there's no obvious way to sort the results of the different queries into one list by relevance (see also - #922)

Particularly with the release of Commerce the need to search both Entries and  Products will become important.  It would be great to be able to specify which element types should be included in the search.

**Comment by angrybrad** [comment]
> Posted by Matt Stein (matt@workingconcept.com) on 2016/04/13 16:12:29 +0000
-----------------------------------

Relevant also for several calendar plugins that use their own Element Types to manage events.

**Comment by timkelty** [comment]
Perhaps this issue should be more concisely renamed to:
"Allow multi-type element queries", the needs for this go beyond just "searching" as the initial request states.

Example: I need to find related elements of _any_ type related to a given element. Without knowing any possible types up front this isn't possible, and even if you do would result in multiple queries.

**Comment by khalwat** [comment]
Yes please. I have a use case where a site has a "search everything", and I have to do two queries, one for entries, and another for users.

**Comment by oddnavy** [comment]
Is this feature on the roadmap? We have a use case where'd we like to search for both commerce products and entries.

**Comment by brandonkelly** [comment]
Not currently. Would be pretty complicated.

**Comment by jamiewade** [comment]
Has there been any advances with this? We are building a site that has Entries, Commerce, and Events (Solspace Calendar), and would love to be able to search through and paginate the results for all of these Elements.

**Comment by nikolowry** [comment]
Having to use Algolia for full-site search is a pain (Entries and Solspace Calendar). This is such an essential piece of core functionality, I really wish it was made a priority.

**Comment by piotrpog** [comment]
Plaease include this functionality! Or at least add option to pass multiple element queries of multiple elements to `paginate` tag, so i can for example mix products and entries and then show them on one paginated list.

**Comment by aaronmclayton** [comment]
I have run into a similar need for this, with Products and any plugins using their own element types, such as Commerce Bundles. To include Bundles with the Products would be a good user experience.
